routes/order.ts

```diff
@@ -45,27 +43,11 @@ const orderRoutes = async (fastify: FastifyInstance) => {
 
     fastify.post("/order", {}, async (request: FastifyRequest<{ Body: BodyType[] }>, reply) => {
         const requestedProducts: BodyType[] = request.body; // Now TypeScript knows that request.body is of type BodyType[]
-        const finalOrder: Order = { id: uuid(), price: 0, products: [], address: "" };
-        const products = await getAllProducts();
-        const ids: number[] = products.map((p) => p.id);
-        try {
-        await Promise.all(requestedProducts.map(async (p) => {
-            if (ids.includes(p.id)) {
-                const matchedProduct = await getProductFromId(p.id); // on reject, catch will be called
-                if (!matchedProduct.quantity) {
-                    throw new Error("Quantity of requested product is 0");
-                }
-                else if (!matchedProduct.isActive) {
-                    throw new Error("Requested product is inactive")
-                }
-                matchedProduct.quantity -= 1;
-                await updateProductFromId(p.id, matchedProduct);
-                finalOrder.products.push(p);
-            }
-        }));
-        orders.push(finalOrder);
-        reply.code(201);
-        reply.send(finalOrder);
+        console.log("Requested Products are:: ", requestedProducts);
+        try{    
+            const finalOrder = await createOrder(requestedProducts);
+            reply.code(201);
+            reply.send(finalOrder);
         } catch (err) {
             throw err;
         }
```

Comment:
Proper error handling is required, on error send 501 status code with a error message caught inside of catch block